api/src/main/java/org/pmiops/workbench/elasticsearch/ElasticFilters.java

```diff
@@ -0,0 +1,395 @@
+      rq = QueryBuilders.rangeQuery("events.start_date");
+      left = mod.getOperands().get(0);
+      if (mod.getOperands().size() > 1) {
+        right = mod.getOperands().get(1);
+      }
+    } else if (ModifierType.AGE_AT_EVENT.equals(mod.getName())) {
+      rq = QueryBuilders.rangeQuery("events.age_at_start");
+      left = Integer.parseInt(mod.getOperands().get(0));
+      if (mod.getOperands().size() > 1) {
+        right = Integer.parseInt(mod.getOperands().get(1));
+      }
+    } else {
+      throw new RuntimeException("modifier is not a date modifier type: " + mod.getName());
+    }
+    switch (mod.getOperator()) {
+      case LESS_THAN:
+        rq.lt(left);
+        break;
+      case GREATER_THAN:
+        rq.gt(left);
+        break;
+      case LESS_THAN_OR_EQUAL_TO:
+        rq.lte(left);
+        break;
+      case GREATER_THAN_OR_EQUAL_TO:
+        rq.gte(left);
+        break;
+      case BETWEEN:
+        rq.gt(left).lt(right);
+        break;
+      case LIKE:
+      case IN:
+      case EQUAL:
+      case NOT_EQUAL:
+      default:
+        throw new BadRequestException("Bad operator for date modifier: " + mod.getOperator());
+    }
+    return rq;
+  }
+
+  private static boolean isStandardConcept(SearchParameter param) {
+    TreeType paramType = TreeType.valueOf(param.getType());
+    return STANDARD_TREES.contains(paramType);
+  }
+
+  private Set<String> toleafConceptIds(List<SearchParameter> params) {
+    Set<String> out = Sets.newHashSet();
+    for (SearchParameter param : params) {
+      if (param.getGroup()) {
+        out.addAll(
+            childrenByCriteriaGroup.get(param)
+                .stream()
+                .map(id -> Long.toString(id))
+                .collect(Collectors.toSet()));
+      } else {
+        out.add(Long.toString(param.getConceptId()));
+      }
+    }
+    return out;
+  }
+
+  private static class FullTreeType {
+    final TreeType type;
+    final TreeSubType subType;
+    private FullTreeType(TreeType type, TreeSubType subType) {
+      this.type = type;
+      this.subType = subType;
+    }
+
+    static FullTreeType fromParam(SearchParameter param) {
+      return new FullTreeType(
+          TreeType.valueOf(param.getType()), TreeSubType.valueOf(param.getSubtype()));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      FullTreeType that = (FullTreeType) o;
+      return type == that.type &&
+          subType == that.subType;
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(type, subType);
+    }
+  }
+
+  private Map<SearchParameter, Set<Long>> buildCriteriaGroupLookup(SearchRequest req) {
+    // Three categories of criteria groups are currently supported in a SearchRequest:
+    // 1. Groups with only tree types specified, e.g. PPI.
```

Comment:
Per Brian discussion make the order match here - this matches PPI survey.

Reply:
doc TODOs:
- SearchParameter.vaue: ICD9/10 codes, deceased, PPI answer, (something else)
- Answers need to be handled in a follow-up
- Add indexing to README and examples of codes